R/neighborhood_ses.R

```diff
@@ -0,0 +1,191 @@
+#'
+#' @section Data source
+#' All SES variables are sourced from [Statistics Canada Census.](https://www12.statcan.gc.ca/census-recensement/2021/ref/98-304/2021001/chap1-eng.cfm)
+#'
+#' The Census of Population provides a detailed and comprehensive statistical portrait of Canada. It’s collected by Statistics Canada conducts once every 5 years, aiming to provide population and dwelling counts for communities of all sizes across Canada.
+#'
+#' On-Marg is a neighborhood-level index showing marginalization differences between areas. It’s created using a portion of Statistics Canada Census variables.
+#' On-Marg has 4 dimensions. There was a name change in 2021 to avoid deficit-based language and better reflect the census measures associated with each dimension
+#' Summary score can be used if all dimensions are in the same direction. If >1 dimensions are in opposite direction, it’s recommended to report those dimensions separately.
+#'
+#' @section Income additional details
+#' Based on theoretical framework, 42 Census variables are selected. After principal component factor analysis, 18 Census variables are used to create 4 ON-Marg dimensions. Each dimension is available in quintiles – Q1 represents the least marginalization, while Q5 represents the most marginalization.
+#'
+#' @section Missing data
+#' This function automatically excludes encounters with missing data and report both the reason and percentage of missingness exclusion
+#'
+#' @param dbcon (`DBIConnection`)\cr
+#' A database connection to any GEMINI database.
+#'
+#' @param census_year (`numeric` | `character`)\cr
+#' Version of census_year chosen. Only 2016 or 2021 are valid inputs.
+#'
+#' @param cohort\cr
+#' Table with all relevant encounters of interest, where each row corresponds to
+#' a single encounter. Must contain GEMINI Encounter ID (`genc_id`).
+#'
+#' @return (`data.frame` | `data.table`)\cr
+#' This function returns a data.table where each row corresponds to a genc_id from the user-provided cohort input, together with the following columns:
+#' - DA (dissemination area) ID from the corresponding census year: da16uid or da21uid. DA is ['a small area composed of one or more neighbouring dissemination blocks and is the smallest standard geographic area for which all census data are disseminated.'](https://www150.statcan.gc.ca/n1/en/catalogue/92-169-X)
+#' - [Visible minority](https://www12.statcan.gc.ca/census-recensement/2016/ref/dict/pop127-eng.cfm): c16_vismin_pct or c21_vismin_pct
+#' - [Neighbourhood-level income](https://www.cihi.ca/sites/default/files/document/toolkit-area-level-measurement-pccf-en.pdf): qnatippe, qnbtippe, qaatippe, qabtippe, atippe, btippe. While qnatippe and qnbtippe are quintiles calculated based on national income distribution, while qaatippe and qabtippe are quintiles constructed separately for each census metropolitan area (CMA), census agglomeration (CA) or residual area within each province.
```

Comment:
I updated the text section about Statistics Canada Census of Population. There are some intricacies about income and immigration, pls see text for details. 

I mentioned long-format questionnaire, I think the 25% households info is enough to hint those variables are less reliable. I don't want to explicitly mention this since Stats Can argues those are representative sample geographically.